sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -695,6 +695,56 @@ abstract class TernaryExpression extends Expression {
   }
 }
 
+/**
+ * A trait resolving nullable, containsNull, valueContainsNull flags of the output date type.
+ * This logic is usually utilized by expressions combining data from multiple child expressions
+ * of non-primitive types (e.g. [[CaseWhen]]).
+ */
+trait NonPrimitiveTypeMergingExpression extends Expression
```

Comment:
`NonPrimitive` -> `Complex`

Reply:
```Complex``` also sounds better to me :-) But I would like to stay consistent of the terminology (see a [comment](https://github.com/apache/spark/pull/20938/files/eeab72786aa0c28b123cc21a87bd8619a63831bd#r181345710) from a different PR.) What about using some neutral name like ```NullFlagsMergingExpression``` or similar?

Reply:
My previous comment seems to be irrelevant since ```complex``` has already been used (```findTypeForComplex```).